src/day23.scala

```diff
@@ -0,0 +1,151 @@
+// using scala 3.1.0
+
+package day23
+
+import scala.util.Using
+import scala.io.Source
+import Position.*
+import scala.annotation.tailrec
+
+@main def part1(): Unit =
+  val answer = part1(readInput())
+  println(s"The answer is: $answer")
+
+@main def part2(): Unit =
+  val answer = part2(readInput())
+  println(s"The answer is:\n$answer")
+
+def readInput(): String =
+  Using.resource(Source.fromFile("input/day23"))(_.mkString)
+
+type Energy = Int
+
+enum Amphipod:
+  case A, B, C, D
+
+  def energy: Energy = this match
```

Comment:
Easy win: make all of these `val` parameters of `enum Amphipod`, so that you don't recompute them all the time.